addons/mail/static/src/model/record.js

```diff
@@ -414,20 +414,69 @@ export class Record {
         return !this.in(collection);
     }
 
-    toData() {
+    toData(options = { depth: false }) {
+        const seenRecords = new Set();
+        const storeData = {};
+        const prefix = this._getPyModelName();
+        let fields = undefined;
+        if (Array.isArray(options)) {
+            fields = options.map((field) => `${prefix}.${field}`);
+        }
+        this._toData(storeData, seenRecords, options.depth, fields, prefix);
+        return storeData;
+    }
```

Comment:
Looks like `storeData, seenRecords, options.depth, fields` represent global data of the ongoing `toData()` in progress.

You could make code look less complex by putting all of them in a single object. Let's think of a name... Not the best probably but here's a quick decent suggestion: `ongoing`?

So signature of internal method becomes:
```js
_toData(ongoing, prefix);
_toDataRelationalRecord(ongoing, prefix);
```

And you can put a typedef for this `ongoing`, so the IDE can help for autocomplete and even describe each part. This could help keeping code look simple as it should.